### ISSUE TITLE
[CTA Section]:QA: Heading levels are not in an hierarchical order

### ISSUE THREAD

**Issue by gopinathan-n** [carbon-design-system/carbon-for-ibm-dotcom#9709]
### Description

Page Element:
CTA section - With content items -Heading hierarchy in the mentioned component 

Issue Description:
Heading levels are not in an hierarchical order. Now the heading levels are in the following order.
H3, H2, H4

Recommend Fix:
As an accessibility best practice, heading levels are expected to be given in an hierarchical order. e.g. H2, H3, H4

User Impact:
Best Practice

Tool:
JAWS_Desktop
Voiceover_iPhone

### Component(s) impacted

CTA Section

### Browser

Safari, Firefox

### Carbon for IBM.com version

v1.40.0-rc.0

### Severity

Severity 4 = The problem is not visible to or noticeable to an average user. Affects minor functionality, no workaround needed.

### Application/website

Carbon for IBM.com

### Package

@carbon/ibmdotcom-web-components

### CodeSandbox example

https://carbon-design-system.github.io/carbon-for-ibm-dotcom/next/web-components https://carbon-design-system.github.io/carbon-for-ibm-dotcom/next/web-components-react

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues
